File: Python/733_flood-fill.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, color: int) -> List[List[int]]:
        xlen = len(image)
        ylen = len(image[0])
        ori = image[sr][sc]
        
        def dfs(x,y):
            if 0<= x < xlen and 0<= y < ylen and image[x][y] ==ori:
                image[x][y] = color
                dfs(x -1,y)
                dfs(x +1,y)
                dfs(x ,y-1)
                dfs(x ,y+1)
                
        if ori != color:
            dfs(sr,sc)
        return image
        

File: Java/733_flood-fill.java
class Solution {
    public int[][] floodFill(int[][] image, int sr, int sc, int color) {

        int original = image[sr][sc];
        if (original != color){
            dfs(image, sr,sc,original, color);
        }
        return image;
    }

    public void dfs(int[][] image, int r, int c,int original,  int color){
        int xLen = image.length;
        int yLen = image[0].length;
        if (0 <=  r  && r < xLen && 0 <= c  && c < yLen && image[r][c] == original){
            image[r][c] = color;
            dfs(image, r+1, c, original, color);
            dfs(image, r, c+1, original, color);
            dfs(image, r-1, c, original, color);
            dfs(image, r, c-1, original, color);
        }
    }
}